proof/proof.go

```diff
@@ -7,10 +7,25 @@ package proof
 
 import (
 	"errors"
+	"hash"
 
 	"github.com/dedis/crypto"
 )
 
+// NOTE This suite is the same as the global one. Writing it is not an easy when you
+// don't know what methods you want to have (i.e.  the same set of methods that
+// edwards25519 implements etc).
```

Comment:
Why isn't answering this question (mostly) as simple as copying the original global Suite as a template (as you did), but then temporarily commenting out all the members of this interface, try to compile it, see what breaks, and uncomment members as you confirm that they're real dependencies?